framework/sandstone.cpp

```diff
@@ -1039,6 +1041,16 @@ int num_packages() {
     return Topology::topology().packages.size();
 }
 
+void reschedule()
+{
+    if (sApp->device_schedule == nullptr) return;
+    int next_cpu = sApp->device_schedule->get_next_cpu();
+    if (!pin_to_logical_processor(LogicalProcessor(cpu_info[0].cpu_number))) {
```

Comment:
You didn't use `next_cpu` here. You probably meant `cpu_info[next_cpu].cpu_number`.

Reply:
Indeed. While writing the selftest I was so confused. lol